PROJECT_GUIDELINES.md

```diff
@@ -0,0 +1,228 @@
+
+5. Code with Method/Prop/Event/Component decorators should have jsdocs. This allows for documentation generation and for better user experience in an editor that has TypeScript intellisense.
+
+### A11y
+
+Components should be accessible and we use the following resources as guides:
+
+- [Google accessibility overview](https://developers.google.com/web/fundamentals/accessibility/)
+- [WAI-ARIA Authoring Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)
+
+### Components
+
+Before the checklist, new components should have an [issue](https://github.com/Esri/calcite-app-components/issues/new?assignees=&labels=new+component%2C+0+-+new%2C+architecture&template=new-component.md&title=New+Component%3A+) so we can determine whether they belong here or calcite-components. If a component does belong here, that's where the [checklist](https://github.com/Esri/calcite-app-components/wiki/New-Component-Checklist) can be used to guide development.
+
+#### General checklist
+
+- Slots are documented at the beginning of each component file after the import statements and before the `@Component` decorator.
+- Please properly mutate and reflect props.
+- Make sure to document props.
+- UI text props start with `text`. For example: `textCollapse` updates the label of the collapse icon when the component is expanded in the calcite action bar.
+- Events are named in present tense and start with the `calciteComponent`. For example: `calciteBlockToggle` is emitted when the header in a calcite block has been clicked.
+- Properties without values are marked as required (with a `!` in front) or optional (`?`). For example: `@Prop() label?: string;` or `@Prop() text!: string;`
+- Set `shadow: true` in the `@Component` options to make sure styles are encapsulated.
+- Naming conventions of properties, methods, and events are consistent between components for consistency.
+- A resource file resides inside each component folder. These contain CSS, ICONS, and SLOTS constants.
+- Internal events are not propagated.
+
+#### Documentation
+
+- We use storybook for a front end view of our doc. In each component folder resides a `component.stories.ts` file.
+- Knobs are added in the stories file so users can edit props dynamically in the storybook UI.
+- Usage folders are in the component folder with a `basic.md` or `advanced.md` file depending on the complexity of the component for notes and snippets that would help the understanding of the component.
+
+#### Best practices
+
+The following resources showcase best practices we follow for our web components:
+
+- [Google Web Component Best Practices](https://developers.google.com/web/fundamentals/web-components/best-practices)
+- [Custom Element Conformance - W3C Editor's Draft](https://html.spec.whatwg.org/multipage/custom-elements.html)
+
+#### Structure
+
+Please see Stencil's [style guide](https://github.com/ionic-team/stencil/blob/master/STYLE_GUIDE.md#file-structure) for details on component structure.
```

Comment:
Suggestion: `We follow Stencil's suggested component structure. See ... for more details.`